reV/SAM/generation.py

```diff
@@ -589,22 +589,39 @@ def ac(self):
         Returns
         -------
         output : np.ndarray
-            1D array of hourly AC inverter power generation in kW.
+            1D array of AC inverter power generation in kW.
             Datatype is float32 and array length is 8760*time_interval.
         """
         return np.array(self['ac'], dtype=np.float32) / 1000
 
     def dc(self):
-        """Get DC array power generation profile (orig timezone) in kW.
+        """
+        Get DC array power generation profile (orig timezone) in kW.
 
         Returns
         -------
         output : np.ndarray
-            1D array of hourly DC array power generation in kW.
+            1D array of DC array power generation in kW.
             Datatype is float32 and array length is 8760*time_interval.
         """
         return np.array(self['dc'], dtype=np.float32) / 1000
 
+    def clipped_power(self):
+        """
+        Get the clipped DC power generated behind the inverter
+        (orig timezone) in kW.
+
+        Returns
+        -------
+        clipped : np.ndarray
+            1D array of clipped DC power in kW.
+            Datatype is float32 and array length is 8760*time_interval.
+        """
+        ac = self.ac()
+        dc = self.dc()
+
+        return np.where(ac < ac.max(), 0, dc - ac)
```

Comment:
Why wouldnt we always want `dc - ac`? Wouldnt we want the clipped power at hour 11 in the image below? 

![image](https://user-images.githubusercontent.com/12700356/100269355-a25fbf00-2f13-11eb-92df-41d04cd9b4ac.png)

Reply:
On the ReEDS side, we want to differentiate between 1) sub-clipping power that can go either to the grid through the inverter or to storage, and 2) clipped power that can only go to storage. Based on the Texas profiles I was looking at (https://www.evernote.com/l/AKpG4LiF_WpJ57R74ygivcCe9ATtWH--6Vo), when multiple NSRDB pixels are averaged you get some pixels that are clipping in a certain hour and some that aren't, so you can't pull out the clipped power. For hour 11, the ratio of `dc`/`ac` power is just 1/efficiency_inverter, which we can model on our side from the `dc` profile for power that goes to the grid.

I realize now that if we do representative rather than average profiles for each reV site, the "some-pixels-are-clipping-while-others-aren't" issue would go away and we could get the clipped profile in hourlize post-processing. But it would still be nice to generate it directly from reV in case someone wants to use average profiles.

Reply:
So we always want to represent clipped power as `np.where(ac < ac.max(), 0, dc - ac)` then? Or is there another term for just the plain `dc - ac` clipping that we should add?

Reply:
Since we already have `dc` and `ac` outputs, we can get `unclipped_dc` from `dc - clipped_power`. I checked and the profiles are the same whether this step is performed before or after the aggregation.